notebooks/ch17/classification_sensitivity_specificity.md

```diff
@@ -0,0 +1,486 @@
+
+Although the goal is to maximize both sensitivity and specificity, they are inversely related so there is a tradeoff. Increasing sensitivity leads to a decrease in specificity and vice versa. For instance, reducing false negatives by lowering the classification threshold increases false positives on data that initially fell near the classification threshold but was correctly labeled with the negative class. 
+
+
+### ROC Curves
+
+A **ROC (Receiver Operating Characteristic) curve** visualizes the tradeoff between sensitivity and specificity by plotting sensitivity and (1 - specificity), also called the false positive rate, at all classification thresholds. A point on the curve represents the sensitivity and false positive rate associated with a specific threshold value. The ROC curve for `words_list_model` is calculated below using scikit-learn's [ROC curve function](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_curve.html):
+
+
+```python
+# This cell will (probably) be visible
+
+from sklearn.metrics import roc_curve
+
+probabilities = words_list_model.predict_proba(X_train)[:, 1]
+false_positive_rate, sensitivity, thresholds = roc_curve(y_train, probabilities, pos_label=1)
+```
+
+
+```python
+# HIDDEN
+
+plt.step(false_positive_rate, sensitivity, color='b', alpha=0.2,
+         where='post')
+plt.fill_between(false_positive_rate, sensitivity, step='post', alpha=0.2,
+                 color='b')
+plt.xlabel('1 - Specificity (false positive rate)')
+plt.ylabel('Sensitivity')
+plt.title('words_list_model ROC Curve')
+```
+
+
+
+
+    Text(0.5,1,'words_list_model ROC Curve')
+
+
+
+
+![png](classification_sensitivity_specificity_files/classification_sensitivity_specificity_24_1.png)
+
+
+Notice that as we move from left to right across the curve, sensitivity increases and the specificity decreases. Generally, the best classification threshold corresponds to high sensitivity and specificity (low false positive rate). Picking a definitive value for the threshold depends on the classification problem and the relative costs of the two error types. Set the classification threshold based on which error type you want to minimize: minimizing false positives maximizes specificity and minimizing false negatives maximizes sensitivity. For email classification, false positives result in important emails being filtered out, so they are worse than false negatives, in which a spam email winds up in the inbox. In medical settings, however, false negatives in a diagnostic test can be much more consequential than false positives.
```

Comment:
I think the last two sentences have a lot of commas that deter from their readability. Here's a suggestion? (I'm honestly not sure how to tackle this; it's wordy but every phrase does matter immensely. It could be that the current text is the best solution possible.)

> In email classification, our goal is to minimize false positives so that important emails do not get filtered out as spam. This is a tradeoff we are willing to make in exchange for higher specificity, in which spam emails may find themselves in the inbox due to a higher false negative rate. However, in a medical setting we prioritize minimizing specificity, since false negatives can be much more consequential than false positives.